internal/pipe/cask/testdata/TestFullPipe/custom_block_url.rb.golden

```diff
@@ -1,20 +1,32 @@
 # This file was generated by GoReleaser. DO NOT EDIT.
 cask "custom_block_url" do
   module GitHubHelper
-    def self.get_asset_api_url(tag, name)
+    def self.github_token
       require "utils/github"
-      release = GitHub.get_release("goreleaser", "example", tag)
-      release["assets"].find { |asset| asset["name"] == name }["url"]
+
+      token = ENV["HOMEBREW_GITHUB_API_TOKEN"]
+      token ||= GitHub::API.credentials
+      raise "Failed to retrieve github api token" if token.nil? || token.empty?
+
+      token
     end
 
-    def self.token
-      require "utils/github"
-      @github_token = ENV["HOMEBREW_GITHUB_API_TOKEN"]
-      unless @github_token
-        @github_token = GitHub::API.credentials
-        raise "Failed to retrieve token" if @github_token.nil? || @github_token.empty?
-      end
-      @github_token
+    def self.release_asset_url(tag, name)
+      require "json"
+      require "net/http"
+      require "uri"
+
+      resp = Net::HTTP.get(
+        URI.parse("https://api.github.com/repos/goreleaser/example/releases/tags/#{tag}"),
+        {
+          "Accept" => "application/vnd.github+json",
+          "Authorization" => "Bearer #{github_token}",
+          "X-GitHub-Api-Version" => "2022-11-28"
+        }
+      )
+
+      release = JSON.parse(resp)
+      release["assets"].find { |asset| asset["name"] == name }["url"]
```

Comment:
If we use ‎`GitHub.get_release`, the GitHub API token retrieval flow is different from that of ‎`GitHubHelper.github_token`.
Therefore, I use a raw HTTP request to align the behavior.